ext/opcache/tests/opcode_store_specified_file.phpt

```diff
@@ -0,0 +1,62 @@
+--TEST--
+Compile php and store opcode to the specified file
+--INI--
+opcache.enable=1
+opcache.enable_cli=1
+opcache.file_cache_only=1
+opcache.optimization_level=-1
```

Comment:
> This would force dependence on the extension versions, php minor versions, value of constants, OS versions, etc. that it was distributed for, **if the distributed files were compiled with optimizations enabled,** making local development harder or causing mysterious crashes due to bad type inference.
> (e.g. `if (\PHP_OS_FAMILY === 'Windows'){...}else{...}` is optimized away by opcache)
> (e.g. `if (extension_loaded('filter'))` is converted to `if(true)` by opcache if the extension is loaded by whatever compiled the bytecode when optimizations are enabled)

I think that allowing optimizations to be applied to files that are intended to be distributed would cause a lot of crashes or edge cases like this.

Even without optimizations, this is still not redistributable, e.g. my comment about the `__FILE__` and `__DIR__` being optimized to the value it was on the environment `/home/distributor/...` that compiled the opcodes instead of the end user's environment such as `/var/www/...` or `/home/enduser/public_html/...`

Reply:
PHP's constants are actually similar to C's macros, and constant values are substituted into the compiled target file at compile time.

This is a problem that every language that needs to be compiled faces